net/core/filter.c

```diff
@@ -5849,6 +5855,278 @@ static const struct bpf_func_proto bpf_tcp_gen_syncookie_proto = {
 
 #endif /* CONFIG_INET */
 
+#ifdef CONFIG_XDP_LUA
+
+static inline void verify_and_lock(void) {
+	struct xdplua_create_work *lw;
+
+	lw = this_cpu_ptr(&luaworks);
+	if (!lw->init) {
+		lw->init = true;
+		spin_lock(&lw->lock);
+	}
+}
+
+BPF_CALL_2(bpf_lua_dataref, struct xdp_buff *, ctx, int, offset) {
```

Comment:
Suggested change, by @lneto
```
BPF_CALL_2(bpf_lua_dataref, struct xdp_buff *, ctx, int, offset) {
	lua_State *L;
        
        if (ctx->data_end <= offset + ctx->data)
		return -1;

	L = xdplua_lock();
	return ldata_newref(L, ctx->data + offset,
			ctx->data_end - ctx->data - offset);
}
```

Reply:
btw, I got your concern about moving such declarations to another file.. but we can solve this using some macro sorcery.. however, we can leave it as a TODO.. let's merge this PR first..